helm/cluster/templates/_helpers.tpl

```diff
@@ -151,4 +151,115 @@ Where `data` is the data to hash and `global` is the top level scope.
 {{- end -}}
 {{- end -}}
 {{- $registry -}}
-{{- end -}}
\ No newline at end of file
+{{- end -}}
+
+{{- define "cluster.internal.get-provider-integration-values" }}
```

Comment:
Will add comments that explain this, but briefly...

Helm templates have a different context, depending on how you call them, from the chart where the template is defined, or from the parent chart for example.

E.g. in cluster chart we have `$.Values.providerIntegration`, if you use it in a template defined in cluster chart and you call that template from the cluster chart as well, then you just use `$.Values.providerIntegration` in the template.

OTOH if you call that same template from a parent chart (e.g. from cluster-aws), then that same Helm value is in `$.Values.cluster.providerIntegration`.

So this helper template here check from where the call has been made, reads either `$.GiantSwarm.providerIntegration` or `$.GiantSwarm.cluster.providerIntegration` and sets that to `$.GiantSwarm.providerIntegration`.

This way we can use `$.GiantSwarm.providerIntegration` in all templates (after calling this helper).

This also simplifies a lot how other shared templates are called, so it's simpler to use them from e.g. cluster-aws and other charts.